packages/fxa-settings/src/models/integrations/base-integration.ts

```diff
@@ -55,18 +48,174 @@ export interface IntegrationFeatures {
   syncOptional: boolean;
 }
 
-/* TODO, do we care about these capabilities/features?
- * -isOpenWebmailButtonVisible: we have a webmail link showing only in desktop v3 on the confirm
- * reset PW page and confirm page. We have this comment: "we do not show [this] in mobile context
- * because it performs worse".
- * -
+export interface RelierSubscriptionInfo {
+  subscriptionProductId: string;
+  subscriptionProductName: string;
+}
+
+export interface RelierClientInfo {
+  clientId: string | undefined;
+  imageUri: string | undefined;
+  serviceName: string | undefined;
+  redirectUri: string | undefined;
+  trusted: boolean | undefined;
+}
+
+/**
+ * A convenience interface for easy mocking / testing. Use this type in components.
  */
+// export interface RelierData {
+//   name: string;
+//   context: string | undefined;
+//   email: string | undefined;
+//   entrypoint: string | undefined;
+//   entrypointExperiment: string | undefined;
+//   entrypointVariation: string | undefined;
+//   resetPasswordConfirm: boolean | undefined;
+//   setting: string | undefined;
+//   service: string | undefined;
+//   style: string | undefined;
+//   uid: string | undefined;
+//   utmCampaign: string | undefined;
+//   utmContent: string | undefined;
+//   utmMedium: string | undefined;
+//   utmSource: string | undefined;
+//   utmTerm: string | undefined;
+
+//   clientInfo: Promise<RelierClientInfo> | undefined;
+//   subscriptionInfo: Promise<RelierSubscriptionInfo> | undefined;
+// }
+
+export interface ResumeTokenInfo {
+  // TBD
+}
+
+// export interface Relier extends RelierUrlData {
+//   getServiceName(): Promise<string>;
+//   getClientInfo(): Promise<RelierClientInfo | undefined>;
+//   isOAuth(): boolean;
+//   isSync(): Promise<boolean>;
+//   shouldOfferToSync(view: string): boolean;
+//   wantsKeys(): boolean;
+//   wantsTwoStepAuthentication(): boolean;
+//   isTrusted(): boolean;
+//   validate(): void;
+//   getService(): string | undefined;
+//   getRedirectUri(): string | undefined;
+// }
+
+export interface RelierAccount {
+  uid: string;
+  email: string;
+  sessionToken: string;
+  verifyIdToken(
+    idTokenHint: string,
+    clientId: string,
+    gracePeriod: number
+  ): Promise<{ sub: string }>;
+  isDefault(): unknown;
+}
+
+export abstract class Integration<
+  T extends IntegrationFeatures = IntegrationFeatures
+> {
+  type: IntegrationType;
+  public features: T = {} as T;
+  // TODO fix data type
+  public data: any;
```

Comment:
@dschom I left a comment on FXA-8099 around some data model organization here. I think `public data` is going to end up something like:
```
type IntegrationData =
  | WebIntegrationData
  | SyncBasicIntegrationData
  | OAuthIntegrationData
  | PairingAuthorityIntegrationData
  | PairingSupplicantIntegrationData;
```

... but I didn't do that here mostly due to time but I didn't want to stick this directly in `base-integration` and have to import it from other files that import from `base-integration` so it feels fine to address in that validation ticket.